### PR TITLE
Update/fix bugs in template for Lens 4.0.0 beta.4

### LINKED ISSUE #9: Update generator template due to upstream Lens API PageRegistration / PageMenuRegistration changes
Related to https://github.com/lensapp/lens/pull/1386

### PR DESCRIPTION
**Pull request by chenhunghan**
- Update the template to reflect changes in Lens' `PageRegistration`/`PageMenuRegistration ` https://github.com/lensapp/lens/pull/1386 https://github.com/lensapp/generator-lens-ext/issues/9
- Fix bugs in eslint configs https://github.com/lensapp/generator-lens-ext/pull/11/commits/04b5cae0d5aa790bc638e97a2cbece3f30c14268
- Update the template to use basic extension APIs, so that extension developer have a good foundation to extend. 
    - In [main.tsx](https://github.com/lensapp/generator-lens-ext/blob/fix/update-template-for-4beta3/generators/app/templates/ext-ts/main.ts) template
      - `LensMainExtension.onActivate`
      -  `LensMainExtension.onDeactivate`
      -  `LensMainExtension.appMenus`
   - In [renderer.tsx](https://github.com/lensapp/generator-lens-ext/blob/fix/update-template-for-4beta3/generators/app/templates/ext-ts/renderer.tsx) template
     - `LensRendererExtension.globalPages`
     - `LensRendererExtension.globalPageMenus`
     - `LensRendererExtension.statusBarItems`
     - `LensRendererExtension.clusterPages`
     - `LensRendererExtension.clusterPageMenus` (include the new sub-menu https://github.com/lensapp/lens/pull/1452 will be in 4.0.0 beta.4) 
- GIF Demo of the extension apis used in the template

demo

- Add inline ASCII graphs in `renderer.tsx` template to help extension developer to get where customer UI element location in Lens.

example:
```
             Lens
       +-----------------------+
       |*|-----|               |
       |*|-----|  <---------------+ clusterPageMenus
       | |-----|               |
       | +-----+               |
       | |     |               |
       | |     |               |
       +-----------------------+
```

Closes https://github.com/lensapp/generator-lens-ext/issues/9

### CODE REVIEW

File: generators/app/templates/ext-ts/main.ts [hunk 1]
```diff
@@ -1,11 +1,115 @@
...
+      console.log("activated");
+
+      // The following block is an example you can do in onActivate()
+      // You can access Lens' state data and periodically logs the name of the 
+      // currently active cluster in Lens.
+      // 
+      // see what else you can do with Store in our API docs
+      // <https://docs.k8slens.dev/master/extensions/api/modules/_core_api_stores_/>
+      const { clusterStore } = Store;
+      this.#timer = setInterval(() => {
```
Comment (jakolehm) [inline generators/app/templates/ext-ts/main.ts]: Interval could be replaced with `reaction` but maybe `setInterval` is easier to understand 🤔

File: generators/app/templates/ext-ts/renderer.tsx [hunk 2]
```diff
@@ -1,16 +1,263 @@
...
+
+  /**
+   *  `globalPages` allows you register custom global page.
+   * 
+   *  The global page is a full-screen page that hides all other content from a window.
+   *
+   *  ```
+   *            Lens
+   *   +-----------------------+
+   *   |*|                     |
```
Comment (jakolehm) [inline generators/app/templates/ext-ts/renderer.tsx]: Does global page also hide left menu?

File: package.json [hunk 3]
```diff
@@ -33,8 +33,15 @@
     "yosay": "^2.0.2"
   },
   "devDependencies": {
+    "@k8slens/extensions": "^4.0.0-beta.3",
```
Comment (jakolehm) [inline package.json]: Probably makes sense bump to beta.4 (it's already in build pipeline)?

File: generators/app/templates/ext-ts/renderer.tsx (hunk 2 above)
Comment (chenhunghan) [reply generators/app/templates/ext-ts/renderer.tsx]: Yeah, You are right

File: generators/app/templates/ext-ts/main.ts (hunk 1 above)
Comment (chenhunghan) [reply generators/app/templates/ext-ts/main.ts]: I think setinterval is easier for average js developer like me

Comment (jim-docker) [reply generators/app/templates/ext-ts/main.ts]: I agree, this is in the main extension guide, keeps it simple to show you can setup/teardown and gives a first, basic exposure to Stores, to show something useful you can do.

File: package.json (hunk 3 above)
Comment (chenhunghan) [reply package.json]: fixed in https://github.com/lensapp/generator-lens-ext/pull/11/commits/aee6a269d01bd898a8a8b7225c4a2e721a87e167

File: generators/app/templates/ext-ts/renderer.tsx (hunk 2 above)
Comment (chenhunghan) [reply generators/app/templates/ext-ts/renderer.tsx]: fixed in https://github.com/lensapp/generator-lens-ext/pull/11/commits/9d3e5011b0c9c38ee61a6ab6eeed0cfe4a65cb18

**Pull request merged**